test/mpi/init/session.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpi.h"
+#include <stdio.h>
+#include <assert.h>
+#include "mpitest.h"
+
+/* This is example 10.8 from MPI Standard 4.0
+ * Simple example illustrating creation of an MPI communicator using the Session
+ * Model. The pre-defined "mpi://WORLD" process set can be used to first create
+ * a local MPI group and then subsequently to create an MPI communicator from
+ * this group.
+ */
+
+int errs = 0;
+
+int library_foo_test(void);
+void library_foo_init(void);
+void library_foo_finalize(void);
+
```

Comment:
Do you want to make the global variables and functions static?

Reply:
The example was to demonstrate how it can be used in separate libraries. Let me see if I can make the global variables static.

Reply:
`int errs` is the only global. I think it is better to stay as global in order to demonstrate the idea of independent library usages.